### PR TITLE
Add new pipelines to the proxy configuration

### PR DESCRIPTION
**Pull request by vickenty**
### What does this PR do?

Update Agent Proxy configuration guide with new intake endpoints. 

### Motivation

Keep docs up-to-date.

### Additional Notes

cc @FlorianVeaux and @AlexandreYang, could you please double-check that agent configuration options in the examples are correct?

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/proxy.md [hunk 1]
```diff
@@ -418,6 +451,12 @@ database_monitoring:
 network_devices:
     metadata:
         dd_url: haproxy.example.com:3841
+    snmp_traps:
+        forwarder:
+            dd_url: haproxy.example.com:3842
+    netflow:
```
Comment (AlexandreYang) [inline content/en/agent/proxy.md:457]: 👋 @vickenty thanks for this PR. Looks good for `snmp_traps`, but for `netflow` it's a bit too early, the feature is not GA yet. I think it's preferred to add `netflow` once it's GA.

Comment (AlexandreYang) [reply content/en/agent/proxy.md:457]: If ok for you, I can create a Jira card on our board to handle that when netflow is GA.

File: content/en/agent/proxy.md [hunk 2]
```diff
@@ -554,11 +593,21 @@ stream {
         proxy_ssl on;
         proxy_pass ndm-intake.{{< region-param key="dd_site" >}}:443;
     }
+    server {
+        listen 3842; #listen for network devices metadata
```
Comment (FlorianVeaux) [inline content/en/agent/proxy.md:597]:
```suggestion
        listen 3842; #listen for network devices traps
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Two spacing suggestions!

File: content/en/agent/proxy.md (hunk 1 above)
Comment (AlexandreYang) [reply content/en/agent/proxy.md:457]: cc @gmarabout

Comment (vickenty) [reply content/en/agent/proxy.md]: Fixed! If you make a card please cc me!

File: content/en/agent/proxy.md [hunk 3]
```diff
@@ -270,6 +270,14 @@ frontend network_devices_metadata_frontend
     option tcplog
     default_backend datadog-network-devices-metadata
 
+# This declares the endpoint where your Agents connects for
+# sending Network Devices SNMP Traps data (e.g the value of "network_devices.snmp_traps.forwarder.dd_url")
+frontend network_devices_metadata_frontend
```
Comment (AlexandreYang) [inline content/en/agent/proxy.md:275]:
```suggestion
frontend network_devices_snmp_traps_frontend
```

?

**Review by AlexandreYang — CHANGES_REQUESTED** [review]

**Review by alai97 — APPROVED** [review]

**Pull request merged**